monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -45,16 +45,33 @@ class HTTPIslandAPIClient(IIslandAPIClient):
     """
 
     def __init__(
-        self, agent_event_serializer_registry: AgentEventSerializerRegistry, http_client: HTTPClient
+        self,
+        agent_event_serializer_registry: AgentEventSerializerRegistry,
+        http_client: HTTPClient,
+        otp: str,
     ):
         self._agent_event_serializer_registry = agent_event_serializer_registry
         self.http_client = http_client
+        self._otp = otp
 
     def connect(
         self,
         island_server: SocketAddress,
     ):
-        self.http_client.connect(island_server)
+        try:
+            self.http_client.set_server(island_server)
```

Comment:
`set_server()` makes a lot of sense as a name for this method. However, it represents temporal coupling.

The original design (call `__init__()` and then call `connect()`) is also temporal coupling. The difference is that this is a pattern people are use to. It's very common for clients to have some kind of temporal coupling, whereas renaming this `set_server()` might obscure the temporal coupling.

I prefer to call it `connect()` for that reason, but `set_server()` is a much more accurate description. I'll leave the choice up to you. Either is fine with me.

Reply:
The reason I renamed the `connect` method is that there's no connect method in HTTP. The interface of this client is that methods in this class represent HTTP methods and methods starting with `set_` sets common parameters. As this is a low-level client I think it's better if it represents HTTP protocol. The higher level client `IIslandHTTPClient` still has the `connect` method.

Reply:
> The reason I renamed the connect method is that there's no connect method in HTTP

Well, that's just plain false! See https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

But that does raise a more important issue: if it were named `connect()`, someone might expect it to perform an HTTP CONNECT, and it doesn't.

>  The interface of this client is that methods in this class represent HTTP methods and methods starting with set_ sets common parameters. As this is a low-level client I think it's better if it represents HTTP protocol.

Ok, this is valid.

Reply:
Taking a look at `HTTPClient`, I think I'd still prefer to bury most of this logic in `set_server()` and say, that `set_server()` is successful if and only if it's able to get("?action=is-up"). This `HTTPClient` is not completely ignorant of the Island, and I think that's ok. I agree it should mostly be low-level HTTP code, but perhaps we can strike a balance. Let's talk over webex. This thread is getting long.